File: services/api/app/models/approvals.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

ApprovalTargetType = Literal[
    "requirement_analysis",
    "task_decomposition",
    "dev_task",
    "subtask",
    "artifact",
    "revision_work_item",
]
ApprovalStatus = Literal["pending", "approved", "rejected", "needs_revision"]


class ApprovalCreate(BaseModel):
    project_id: str
    target_type: ApprovalTargetType
    target_id: str
    feedback: str | None = None


class ApprovalUpdate(BaseModel):
    status: ApprovalStatus
    feedback: str | None = None


class Approval(BaseModel):
    id: str
    project_id: str
    target_type: ApprovalTargetType
    target_id: str
    status: ApprovalStatus
    requested_by: str
    decided_by: str | None = None
    feedback: str | None = None
    created_at: datetime
    updated_at: datetime
    decided_at: datetime | None = None

File: apps/web/src/types/approvals.ts
export type ApprovalTargetType =
  | 'requirement_analysis' | 'task_decomposition' | 'dev_task' | 'subtask' | 'artifact'
export type ApprovalStatus = 'pending' | 'approved' | 'rejected' | 'needs_revision'

export interface ApprovalCreate {
  project_id: string
  target_type: ApprovalTargetType
  target_id: string
  feedback?: string | null
}

export interface ApprovalDecision {
  status: 'approved' | 'rejected' | 'needs_revision'
  feedback?: string | null
}

export interface Approval {
  id: string
  project_id: string
  target_type: ApprovalTargetType
  target_id: string
  status: ApprovalStatus
  requested_by: string
  decided_by: string | null
  feedback: string | null
  created_at: string
  updated_at: string
  decided_at: string | null
}
